### ISSUE TITLE
Training Steps per Epoch Chapter 6.3

### ISSUE THREAD

**Issue by WillKoehrsen** [fchollet/deep-learning-with-python-notebooks#22]
The training generator is set up with a batch size of 128, and when fitting the models, the code uses 500 steps per training epoch. 

```python
history = model.fit_generator(train_gen,
                              steps_per_epoch=500,
                              epochs=20,
                              validation_data=val_gen,
                              validation_steps=val_steps)
```

Doesn't this mean the model is only training on 128 * 500 = 64000 training data points rather than all 200000? For validation, the steps per validation epoch are set to val_steps where

```python
# This is how many steps to draw from `val_gen`
# in order to see the whole validation set:
val_steps = (300000 - 200001 - lookback) // batch_size
```
I probably am misunderstanding some part of this example, but why are we not using all the training examples every epoch? It seems like we could define the number of training steps as

```python
train_steps = (200000 - lookback) // batch_size)
```
in order to use all the training points every epoch.

**Comment by miguel-data-sc** [comment]
I would say that's an error in the notebook (also present in  code for R version, btw)

It is not exactly  training limited to 64000 data points but limited **per epoch**. This is, as samples are being shuffled all 200.000 training data points are selected randomly throughout different  epochs, with -many- repeated before others are randomly selected. 

In practice this should make training unstable and reduce the capacity of the net to fit or deliberate overfit. 

This is my understanding, wiser opinions will be welcome.

**Comment by ChenZhongPu** [comment]
I have the same concern. Is it a mistake here?
